diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -534,7 +560,6 @@ rcl_interfaces::msg::SetParametersResult DiffDriveController::on_param_change(co
   {
```

Comment:
#this is wrong as u see before for iteration its initialize as false the result.succesfull, Therefore, when its get in to "if" stats its returning true, in your's scenerio its always returning true. So whats the point ? @Zarqu0n

Reply:
We change the all parameters in here https://github.com/saha-robotics/ros2_controllers/blob/3e2a5c052761887facc56127d5a7992f99eed3c3/diff_drive_controller/src/diff_drive_controller.cpp#L132-L155

If I were to do as you said, I would have to check all the parameters one by one in this callback and return if the parameter was entered and its name was entered. I did not do this because I was lazy and because it would not look bad in the code line. In its current state, even if a false parameter is returned, it will return true, but it will not change any parameter value.

Reply:
If its your choice okay then,